tests/unit/test_charm_leader_unit.py

```diff
@@ -0,0 +1,234 @@
+#!/usr/bin/env python3
+# Copyright 2022 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+import json
+import unittest
+from unittest.mock import Mock, PropertyMock, call, patch
+
+import yaml
+from ops import testing
+from ops.model import ActiveStatus, BlockedStatus, WaitingStatus
+
+from charm import AlertmanagerConfigurerOperatorCharm
+
+TEST_MULTITENANT_LABEL = "some_test_label"
+TEST_CONFIG = f"""options:
+  multitenant_label:
+    type: string
+    description: |
+      Alertmanager Configurer has been designed to support multiple tenants. In a multitenant
+      Alertmanager Configurer setup, each alert is first routed on the tenancy label, and then
+      the routing tree is distinct for each tenant.
+    default: {TEST_MULTITENANT_LABEL}
+"""
+with open("./tests/unit/test_config/alertmanager_default.yml", "r") as default_yaml:
+    TEST_ALERTMANAGER_DEFAULT_CONFIG = default_yaml.read()
+TEST_ALERTMANAGER_CONFIG_FILE = "/test/rules/dir/config_file.yml"
+ALERTMANAGER_CLASS = "charm.AlertmanagerConfigurerOperatorCharm"
+
+
+class TestAlertmanagerConfigurerOperatorCharmLeader(unittest.TestCase):
+    @patch("charm.KubernetesServicePatch", lambda charm, ports: None)
+    def setUp(self):
+        testing.SIMULATE_CAN_CONNECT = True
+        self.harness = testing.Harness(AlertmanagerConfigurerOperatorCharm, config=TEST_CONFIG)
+        self.addCleanup(self.harness.cleanup)
+        self.harness.set_leader(True)
+        self.harness.begin()
+
+    @patch("charm.AlertmanagerConfigDirWatcher")
+    @patch(f"{ALERTMANAGER_CLASS}.ALERTMANAGER_CONFIG_DIR", new_callable=PropertyMock)
+    @patch("ops.model.Container.push", Mock())
+    def test_given_alertmanager_config_directory_when_start_then_watchdog_starts_watching_given_alertmanager_config_directory(  # noqa: E501
```

Comment:
First one was from the `given-when-then` and the second one was just a word - watchdog starts watching given/specified directory. 
I removed the second given to avoid confusion.